addons/website_membership/controllers/main.py

```diff
@@ -89,57 +85,55 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=1, *
 
         google_map_partner_ids = []
-        if request.env.ref('website_membership.opt_index_google_map').customize_show:
-            membership_lines_ids = membership_line_obj.search(cr, uid, line_domain, context=context)
-            google_map_partner_ids = membership_line_obj.get_published_companies(cr, uid, membership_line_ids, limit=2000, context=context)
+        if request.env.ref('website_membership.opt_index_google_map').customize_show and membership_id == 'free':
+            membership_lines = request.env['membership.membership_line'].search(line_domain)
+            google_map_partner_ids = membership_lines.get_published_companies(limit=2000)
 
         search_domain = [('membership_state', '=', 'free'), ('website_published', '=', True)]
         if post_name:
             search_domain += ['|', ('name', 'ilike', post_name), ('website_description', 'ilike', post_name)]
         if country_id:
             search_domain += [('country_id', '=', country_id)]
-        free_partner_ids = partner_obj.search(cr, SUPERUSER_ID, search_domain, context=context)
+        free_partners = Partner.sudo().search(search_domain).ids
+
         memberships_data = []
         for membership_record in memberships:
             memberships_data.append({'id': membership_record.id, 'name': membership_record.name})
         memberships_partner_ids = {}
-        for line in membership_lines:
-            memberships_partner_ids.setdefault(line.membership_id.id, []).append(line.partner.id)
-        if free_partner_ids:
+        if membership_id != 'free':
+            for line in membership_lines:
+                memberships_partner_ids.setdefault(line.membership_id.id, []).append(line.partner.id)
+        if free_partners:
             memberships_data.append({'id': 'free', 'name': _('Free Members')})
             if not membership_id or membership_id == 'free':
                 if count_members < offset + limit:
                     free_start = max(offset - count_members, 0)
                     free_end = max(offset + limit - count_members, 0)
-                    memberships_partner_ids['free'] = free_partner_ids[free_start:free_end]
-                    page_partner_ids |= set(memberships_partner_ids['free'])
-                google_map_partner_ids += free_partner_ids[:2000-len(google_map_partner_ids)]
-                count_members += len(free_partner_ids)
+                    memberships_partner_ids['free'] = free_partners[free_start:free_end]
+                    partner_ids |= set(memberships_partner_ids['free'])
+                google_map_partner_ids += free_partners[:2000-len(google_map_partner_ids)]
+                count_members += len(free_partners)
 
         google_map_partner_ids = ",".join(map(str, google_map_partner_ids))
 
-        partners = { p.id: p for p in partner_obj.browse(request.cr, SUPERUSER_ID, list(page_partner_ids), request.context)}
+        partners = {p.id: p for p in Partner.sudo().browse(list(partner_ids))}
```

Comment:
This browse can be removed if we use mapped at [line#102](https://github.com/odoo/odoo/pull/5420/files#diff-ca09ccdcee1d1858d12146bb9f0d04f5R102)